djangocms_versioning/models.py

```diff
@@ -102,9 +102,13 @@ def __str__(self):
         return "Version #{}".format(self.pk)
 
     def save(self, **kwargs):
+        from .operations import send_post_version_operation, send_pre_version_operation
+
         created = not self.pk
         # On version creation
         if created:
+            # trigger pre operation signal
+            action_token = send_pre_version_operation(constants.DRAFT, obj=self)
```

Comment:
Think we've discussed that this signal (on version creation) is not needed, because pre/post_save signals are already there

Reply:
I would argue that the exisiing standard signal should be disabled and this custom one is kept as it keeps them consistent (draft would be different) and makes them easier to reuse.

Reply:
Disabling standard django pre_save and post_save signals is likely to not be straightforward. In terms of keeping consistency I think it's more important to keep things consistent with django and django-cms (which doesn't disable any standard django signals AFAIK) than with versioning which is a library most developers will know less about than django and django-cms.

Reply:
@monikasulik I agree to not trying to remove them as I originally said. I don't agree with using the Django default signals just for the DRAFT creation as they will contain a slightly different structure and will be different compared to the Versioning supplied ones other than DRAFT i.e. ARCHIVE, PUBLISH, UNPUBLISH.

What are your thoughts?

Reply:
My thoughts atm are to follow YAGNI :D https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it I do feel draft is a different case as we never actually "change" to it. But maybe if you explain how you intend to be using these signals in the migration project I will change my mind ;)

Reply:
That's a terrible principle. Don't implement performance increases until you need them, in which case it's much harder to refactor rather than bake it in.

Reply:
Actually, the one time I was on a project that religiously followed YAGNI, it was the easiest project I've ever had to maintain. And that despite the fact that it had a very similar architecture to what we've been doing on gaps (i.e. a constantly growing number of repos that have to work together).
Besides, performance issues are much more likely to happen when something is over-engineered than under-engineered. And when a performance issue does happen in code that's under-engineered it tends to be much easier to correct (it'll be something like adding select_related somewhere vs. spending hours figuring out where exactly an issue is happening in lots of code that was made more abstract that it needed to be).
And also, if you know you'll need the performance increase (as we do when working on gaps) then it's not breaking YAGNI to have it in.